src/Arriba/Arriba/Diagnostics/Observability/ArribaObservationContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Arriba.Diagnostics.Observability
+{
+    public class ArribaObservationContext : IObservationContext
+    {
+        Guid _contextId;
+        Guid _sequenceId;
+
+        public Guid ContextId => _contextId;
+
+        public Guid SequenceId => _sequenceId;
+
+        public long TimeStamp { get; }
+
+        public ArribaObservationContext() 
+        {
+            _contextId = new Guid();
+            _sequenceId = _contextId;
+            TimeStamp = DateTime.Now.Ticks;
+        }
+
+        public ArribaObservationContext(IObservationContext context)
```

Comment:
duplicate constructors is a source of bugs. introduce a private or protected constructor which takes the Guid() of the contextId then ensure each of these calls that constructor
```suggestion
        public ArribaObservationContext(IObservationContext context)
                : this(context.ContextId)
                {}
  
        private ArribaObservationContext(Guid contextId)              
                
```

Reply:
Thanks, refactored to have private constructors called